File: testsuite/Jogl4Examples/ShadersExample/src/ShadersMouseInteractionTechniques.java
import java.awt.event.MouseEvent;
import java.awt.event.MouseWheelEvent;

import com.jogamp.opengl.awt.GLCanvas;

import vsdk.toolkit.gui.AwtSystem;

public class ShadersMouseInteractionTechniques
{
    public void processMouseEntered(GLCanvas canvas)
    {
        if ( canvas != null ) {
            canvas.requestFocusInWindow();
        }
    }

    public void processMouseExited()
    {
    }

    public boolean processMousePressed(ShadersModel model, MouseEvent e)
    {
        return model.getCameraController().processMousePressedEvent(AwtSystem.awt2vsdkEvent(e));
    }

    public boolean processMouseReleased(ShadersModel model, MouseEvent e)
    {
        return model.getCameraController().processMouseReleasedEvent(AwtSystem.awt2vsdkEvent(e));
    }

    public boolean processMouseClicked(ShadersModel model, MouseEvent e)
    {
        return model.getCameraController().processMouseClickedEvent(AwtSystem.awt2vsdkEvent(e));
    }

    public boolean processMouseMoved(ShadersModel model, MouseEvent e)
    {
        return model.getCameraController().processMouseMovedEvent(AwtSystem.awt2vsdkEvent(e));
    }

    public boolean processMouseDragged(ShadersModel model, MouseEvent e)
    {
        return model.getCameraController().processMouseDraggedEvent(AwtSystem.awt2vsdkEvent(e));
    }

    public boolean processMouseWheelMoved(ShadersModel model, MouseWheelEvent e)
    {
        return model.getCameraController().processMouseWheelEvent(AwtSystem.awt2vsdkEvent(e));
    }
}

File: cpp/testsuite/OpenGL4Examples/ShadersExample/src/gui/ShadersMouseInteractionTechniques.cpp
#include "ShadersMouseInteractionTechniques.h"
#include "../model/ShadersModel.h"

bool ShadersMouseInteractionTechniques::processMousePressed(ShadersModel* model, const MouseEvent& e) { return model && model->cameraController->processMousePressedEvent(e); }
bool ShadersMouseInteractionTechniques::processMouseReleased(ShadersModel* model, const MouseEvent& e) { return model && model->cameraController->processMouseReleasedEvent(e); }
bool ShadersMouseInteractionTechniques::processMouseClicked(ShadersModel* model, const MouseEvent& e) { return model && model->cameraController->processMouseClickedEvent(e); }
bool ShadersMouseInteractionTechniques::processMouseMoved(ShadersModel* model, const MouseEvent& e) { return model && model->cameraController->processMouseMovedEvent(e); }
bool ShadersMouseInteractionTechniques::processMouseDragged(ShadersModel* model, const MouseEvent& e) { return model && model->cameraController->processMouseDraggedEvent(e); }
bool ShadersMouseInteractionTechniques::processMouseWheelMoved(ShadersModel* model, const MouseEvent& e) { return model && model->cameraController->processMouseWheelEvent(e); }
bool ShadersMouseInteractionTechniques::processMousePressedForApp(CameraControllerAquynza* cameraController, const MouseEvent& e) { return cameraController && cameraController->processMousePressedEvent(e); }
bool ShadersMouseInteractionTechniques::processMouseReleasedForApp(CameraControllerAquynza* cameraController, const MouseEvent& e) { return cameraController && cameraController->processMouseReleasedEvent(e); }
bool ShadersMouseInteractionTechniques::processMouseMovedForApp(CameraControllerAquynza* cameraController, const MouseEvent& e) { return cameraController && cameraController->processMouseMovedEvent(e); }
bool ShadersMouseInteractionTechniques::processMouseDraggedForApp(CameraControllerAquynza* cameraController, const MouseEvent& e) { return cameraController && cameraController->processMouseDraggedEvent(e); }
bool ShadersMouseInteractionTechniques::processMouseWheelMovedForApp(CameraControllerAquynza* cameraController, const MouseEvent& e) { return cameraController && cameraController->processMouseWheelEvent(e); }
